aten/src/ATen/core/TensorBase.h

```diff
@@ -301,7 +301,7 @@ class TORCH_API TensorBase {
     return impl_->storage();
   }
   bool is_alias_of(const at::TensorBase& other) const{
-    return impl_->storage().is_alias_of(other.storage());
+    return impl_->is_alias_of(*other.unsafeGetTensorImpl());
```

Comment:
This function is used in perf sensitive places so not sure that adding the virtual call is ok here.
```cpp
return impl_->has_strorage() && impl_->storage().is_alias_of(other.storage());
```
should give you the same feature without the perf loss right?

Reply:
@albanD for backends with lazy tensors like HPU, we still retain the view tensor notion internally so that they share the same storage with the original tensor in the device and updating a view tensor results in an update on the original tensor. However, the lazy tensors are storage less and impl_->has_storage() will always return false even if the two tensors are views.
We want a way to override the default is_alias_of so that we can define it based on the view related records we maintain in our bridge. I did notice that the is_alias_of gets used in the DDP code and making it virtual may have a slightly higher cost, but currently the DDP code seems to assume that the tensors are with storage always. Hence, I had to make it virtual so that we can override the default behavior for storage less tensors.
Is there any other suggestion from you about how this can be done for storage less tensors to return true for is_alias_of for views like gradient as bucket view cases in DDP?

Reply:
The DDP usage is not the one that worries me. It is more the use within the jit for IValues as well as the use in TensorIterator (which is the base of all our low level kernels).

If your Tensors have a concept of "storage" and actually do share them. Why not populate it here so that this kind of logic will work as expected?

Reply:
I had noticed the usage in TorchScript IValue, but didn't notice the use in TensorIterator. Is the TorchScript JIT in the time sensitive path? There are a lot of asserts in the autograd generated ops where the storage is_alias_of() is checked if the tensor has_storage() is true. They sounded fine as the code is explicitly checking for storage presence of tensors. 
  Can you point me to the TensirIterator usage? If it is in performance sensitive code, but does not handle storageless lazy tensors, maybe there should be another API that can be used for possibly storageless tensors explicitly? Let me know your opinion. How is it going to be handled for lazy tensor core, will is_alias_of() work for them?
  For the tensors in our backend, they are lazy tensors and do not have storage. Hence, at the PyTorch at::tensor level the has_storage() is false. I was mentioning that internally, we do bookeeping for views created on tensors and on the device they share storage, but this is handled within the backend during tensor allocations and not associated with the lazy tensor.

Reply:
We had discussions with @wconstab about the HPU lazy tensor and the current LazyTensorCore. There are few differences, including the functionalization pass, so it isn't usable for HPU in the current form.

Reply:
> Can you point me to the TensirIterator usage?

There is one in TensorIteratorReduce.cpp and MemoryOverlap.cpp (which is called to check if ops are valid IIRC).
Did you discuss with @wconstab how lazy tensor handles it?

Reply:
@albanD , thanks - I had seen the TensorIteratorReduce.cpp, but was not sure if it is important from performance point of view as it is inside an assert. For the MemoryOverlap.cpp, it is calling the is_alias_of() from the Storage() class directly, if the storage is available. This should be unchanged with this PR, as I am changing the TensorBase is_alias_of(). Shall I modify the TensorIteratorReduce.cpp to similarly use the Storage->is_alias_of(), will it be fine?

We didn't discuss about this particular feature for LazyTensorCore with @wconstab , we had a discussion on the high level differences between HPU lazy tensor and XLA lazy tensor approach. I will discuss this with him.